sdk/management/docs/MIGRATION_GUIDE.md

```diff
@@ -0,0 +1,306 @@
+
+**In old version (`com.microsoft.azure.management.**`)**
+
+```java
+OkHttpClient.Builder builder = new OkHttpClient.Builder().proxy(new Proxy(Proxy.Type.HTTP, new InetSocketAddress("127.0.0.1", 8888)));
+RestClient client = new RestClient.Builder(builder, new Retrofit.Builder())
+    .withCredentials(credential)
+    .build();
+
+Azure azure = Azure.authenticate(client, "<TenantId>")
+    .withDefaultSubscription();
+```
+
+**Equivalent in new version (`com.azure.resourcemanager.**`)**
+
+```java
+HttpClient client = new OkHttpAsyncHttpClientBuilder()
+    .proxy(new ProxyOptions(ProxyOptions.Type.HTTP, new InetSocketAddress("127.0.0.1", 8888)))
+    .build();
+
+Azure azure = Azure.configure()
+    .withHttpClient(client)
+    .authenticate(credential, profile)
+    .withDefaultSubscription();
+```    
+    
+## Error Handling
+
+There is a minor namespace change in the exception class. To be specific, the previous `CloudException` has been re-named to `ManagementException`. 
+
+**In old version (`com.microsoft.azure.management.**`)**
+
+```java
+final String resourceGroupName = "invalid resource group name";
+try {
+    azure.resourceGroups().define(resourceGroupName)
+        .withRegion(Region.US_WEST2)
+        .create();
+} catch (CloudException e) {
+    System.err.printf("Response code: %s%n", e.body().code());
+    System.err.printf("Response message: %s%n", e.body().message());
+}
+```
+
+**Equivalent in new version (`com.azure.resourcemanager.**`)**
+
+```java
+final String resourceGroupName = "invalid resource group name";
+try {
+    azure.resourceGroups().define(resourceGroupName)
+        .withRegion(Region.US_WEST2)
+        .create();
+} catch (ManagementException e) {
+    System.err.printf("Response code: %s%n", e.getValue().getCode());
+    System.err.printf("Response message: %s%n", e.getValue().getMessage());
+}
+```
+
+## rxJava to Reactor
+
+In old version (`com.microsoft.azure.management.**`), `rxJava` is used for non-blocking applications
+
+In new version (`com.azure.resourcemanager.**`), we have adopted `Reactor` as the main library in replacement of `rxJava` due to the Azure Core adoption.
+
+**In old version (`com.microsoft.azure.management.**`)**
+
+```java
+azure.publicIPAddresses()
+    .define(publicIpName)
+    .withRegion(region)
+    .withExistingResourceGroup(rgName)
+    .withLeafDomainLabel(publicIpName)
+    .createAsync().flatMap(new Func1 < Indexable, Observable < Indexable >> () {
+        @Override
+        public Observable < Indexable > call(Indexable indexable) {
+            if (indexable instanceof PublicIPAddress) {
+                PublicIPAddress publicIp = (PublicIPAddress) indexable;
+                //=============================================================
```

Comment:
Do we need to keep the comments and lines of `println()` here? This section is to introduce how the conversion from rxJava to Reactor would be.

Reply:
good point